geoparquet/index.qmd

```diff
@@ -0,0 +1,86 @@
+
+### Column filtering
+
+Because Parquet is column oriented, a Parquet reader can fetch only specific columns that the user is interested in.
+
+### Row group filtering
+
+Because Parquet is internally chunked, Parquet can fetch only specific row groups that meet a specific filtering condition.
+
+Note that row group filtering on a specific column tends to only work well if the Parquet file was sorted on that column when saved. Non-sorted columns tend to have random values, and so the column statistics won't tend to filter out many row groups.
+
+### Internal compression by default
+
+Parquet is internally compressed _by default_. Additionally, Parquet compression tends to be very efficient, leading to much smaller file sizes than comparable formats.
+
+Compression algorithms are more effective when nearby bytes are more similar to each other. Data within a column tends to be much more similar than data across a row. Since Parquet is column oriented, compression algorithms work better and result in smaller file sizes than on a comparable row-based format.
+
+It's possible to have random access to one of the internal chunks inside the file at large, even though that chunk is compressed. Note that it isn't possible to fetch partial data _inside one chunk_ without loading and decompressing the entire chunk.
+
+### Geometries encoded as Well-Known Binary
+
+For maximum compatibility with existing systems, geometries are stored as [ISO-standard WKB](https://github.com/opengeospatial/geoparquet/blob/main/format-specs/geoparquet.md#encoding). Most geospatial programs are able to read and write WKB.
+
+### No spatial index (yet!)
+
+GeoParquet is a young specification, and spatial indices are not yet part of the standard. Future revisions of GeoParquet are expected to add support for spatial indexes.
+
+One way around this is to store multiple GeoParquet files according to some region identifier, cataloging each file with STAC.
+
+### Multi-file support
+
+While at medium data sizes GeoParquet is most easily distributed as a single file, at large data sizes a single dataset is often split into multiple files. Sometimes multiple files can be easier to write, such as if the data is output from a distributed system.
+
+### Multithreading support but no streaming
+
+In a streaming download, you read bytes starting at the beginning of the file, progressing towards the end. In Parquet, this is not helpful because the metadata is in the _footer_ of the file instead of the header.
+
+Instead, we can replicate something similar to streaming by first fetching only the metadata region at the end of the file, and then making multiple requests for each internal chunk.
+
+### Files are immutable
+
+Once written, a Parquet file is immutable. No modification or appending can happen to that Parquet file. Instead, create a new Parquet file.
```

Comment:
For Parquet stored as multi-files, would it be possible to 'append' new rows by adding a new chunk file?

Reply:
Isn't there a main file that would need to be updated to know about the new chunk?

Reply:
> For Parquet stored as multi-files, would it be possible to 'append' new rows by adding a new chunk file?

Yes, as long as it has the same schema.

> Isn't there a main file that would need to be updated to know about the new chunk?

It depends. There _can_ be a metadata file at the top level in the folder that describes all the Parquet files in the directory. That file doesn't always exist (it's not required by the parquet spec; the parquet spec is mostly concerned with individual files, and people build support for multiple files on top of it).

But it's relatively easy to update that top-level metadata file, compared with creating a new Parquet file.